actions/contacts/upsert.go

```diff
@@ -0,0 +1,60 @@
+package contacts
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/bitcoin-sv/spv-wallet/engine"
+	"github.com/bitcoin-sv/spv-wallet/mappings"
+	"github.com/bitcoin-sv/spv-wallet/models"
+	"github.com/bitcoin-sv/spv-wallet/server/auth"
+	"github.com/gin-gonic/gin"
+)
+
+// upsert will add a new contact or modify an existing one.
+// Upsert contact godoc
+// @Summary		Upsert contact
+// @Description	Add or update contact. For new contact send request to add current user as contact
+// @Tags		Contact
+// @Produce		json
+// @Param		UpsertContact body contacts.UpsertContact true "Full name and metadata needed to add/modify contact"
```

Comment:
I would insert path param like this:
// @Param		paymail path string true "some desc"
and regenerate swagger

Reply:
fixed